mlir/lib/Pass/PassManagerOptions.cpp

```diff
@@ -58,6 +58,12 @@ struct PassManagerOptions {
       llvm::cl::desc("When printing IR for print-ir-[before|after]{-all} "
                      "always print the top-level operation"),
       llvm::cl::init(false)};
+  llvm::cl::opt<bool> printSkipRegions{
+      "mlir-print-ir-skip-regions",
```

Comment:
The name of this option does not match a PassManager flag, it looks like a more general option like `mlir-print-op-generic`: why didn't you implement it there?

Reply:
The AsmPrinter flags seem to have a global effect, I this would disallow me from using `--mlir-print-ir-*` in the concise form with verbose prints (inc., regions) with `--debug-only=`, no? I would like this combination to be allowed.
I'm not very familiar with this part of the codebase so please correct me if I'm wrong.

Reply:
> I would like this combination to be supported.

Which combination? Why would the "pass manager" using a concise form when it wouldn't be used elsewhere? That seems somehow arbitrary here.

Also I mostly commented on the option name which does not reflect the behavior.

Reply:
I see your point. It's not exactly what I had in mind, but it does seem more consistent considering the existing options in AsmPrinter.

I moved the flag to `AsmPrinter` and renamed it to `--mlir-print-skip-regions`.